libvast/vast/concept/convertible/data.hpp

```diff
@@ -413,14 +422,33 @@ caf::error convert(const record& src, To& dst) {
   return convert(src, dst, dst.layout);
 }
 
+// TODO: Move to a dedicated header after conversion is refactored to use
+// specialization.
```

Comment:
Please expand on this comment. Specialization for what?

Reply:
With specialization I mean that we must  use specializations of `template <class, class, class> converter;` instead of function overloading of `convert()` so we can write a concept (or detector) that actually works.